posixlib/src/main/scala/scala/scalanative/posix/locale.scala

```diff
@@ -3,14 +3,103 @@ package posix
 
 import scala.scalanative.unsafe._
 
-@extern
-object locale {
+@extern object locale extends locale {}
```

Comment:
Eric,  

Thank you for the review & suggestion.

1) The intention of the code you suggested changing is for the posix locale object to
     extend the posix locale trait (which extends libc.locale trait but also has other stuff).
    I believe the code as is does that. After study, I believe the suggested replacement
    does not.

    I have not tried it but perhaps something like `@extern object locale extends posix.locale {}`
   might make the intention clearer.  Is there something better?

2) I like starting off restrictive and then possibly loosening up as time delivers more information.
    Here, I would avoid overthinking things, let the situation be and the experiment play out.
    Yes, there is possibly someone out there who can hurt themselves, but there probably are
    not two.

    While "First do no harm" is always true, we have a severely limited amount of development
    and review time for posixlib & clib (and a putative oslib) and a __lot__ of technical debut.
    I think the current 'extern' situation is reasonably adequate.

Gotta run,  appointment here.  Sorry.  More later.

Reply:
For 1. above it would certainly work because that is how libc Math.scala works. We shouldn't need the traits at all in posix as we just want to reuse the libc traits.

For 2. above, I was saying that we may not need posix traits at all unless we plan to allow downstream developers to use them. We could make the trait in clib `private [scalanative] trait math` for example if we only want to available for posix usage.

Reply:
Eric, I changed the code as you described in #1 above. I do think the new code makes
more sense while still fulfilling the intent of the original code.

As for #2, I am not adverse to changing the clib traits to `private [scalanative]` but do 
not think it is the best use of time for now and the near-to-mid future.

Reply:
As for 2) above, I was hoping @WojciechMazur might weigh in. Having the public traits allows developers to combine and use them in strange ways. We may want to not allow that to save us questions and debugging in the middle of the night. We don't have a valid use case or request for that feature either so we might be opening ourselves up to pain so locking that down might be prudent.

FWIW, I didn't think of that earlier.